src/libAtomVM/term.h

```diff
@@ -87,6 +87,7 @@ extern "C" {
 #define TUPLE_SIZE(elems) ((int) (elems + 1))
 #define CONS_SIZE 2
 #define REFC_BINARY_CONS_OFFSET 4
+#define LIST_SIZE(num_elements, element_size) (num_elements * (element_size + CONS_SIZE))
```

Comment:
Further nitpicking (this might save us a lot of headache).
Macro should be rewritten as: `((num_elements) * ((element_size) + CONS_SIZE))`.
If you do something like: `LIST_SIZE(a_elements + b_elements, TUPLE_SIZE(3))` it will expand as: `(a_elements + b_elements * (TUPLE_SIZE(3) + CONS_SIZE))`.
I can assure that debugging this is very painful.

Reply:
I don't think it's nitpicking at all, and I should have picked up on this first, so my bad.  Nice catch.